src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -83,6 +86,7 @@ PocoHTTPClient::PocoHTTPClient(const PocoHTTPClientConfiguration & clientConfigu
           Poco::Timespan(clientConfiguration.httpRequestTimeoutMs * 1000) /// receive timeout.
           ))
     , remote_host_filter(clientConfiguration.remote_host_filter)
+    , global_context(clientConfiguration.global_context)
```

Comment:
Maybe we can pass only settings without the whole context?

Reply:
Fixed